plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -365,20 +389,24 @@ static void noise_suppress_update(void *data, obs_data_t *s)
 
 #ifdef LIBNVAFX_ENABLED
 	ng->intensity_ratio = (float)obs_data_get_double(s, S_NVAFX_INTENSITY);
-	if (ng->nvafx_initialized && ng->use_nvafx) {
-		int err;
-		for (int i = 0; i < MAX_PREPROC_CHANNELS; i++) {
-			err = NvAFX_SetFloat(
-				ng->handle[i],
-				NVAFX_PARAM_DENOISER_INTENSITY_RATIO,
-				ng->intensity_ratio);
-			if (err != NVAFX_STATUS_SUCCESS) {
-				do_log(LOG_ERROR,
-				       "NvAFX_SetFloat(Intensity Ratio: %f) failed, error %i",
-				       ng->intensity_ratio, err);
-				ng->use_nvafx = false;
+	if (ng->use_nvafx) {
+		pthread_mutex_lock(&ng->nvafx_mutex);
```

Comment:
The nvafx initialization thread holds this lock while it's running, so trying to acquire the lock here blocks the filter from updating until nvafx is loaded. I believe this is where the issues are stemming from.

Reply:
The intent was to block if a user added the filter and then immediately adjusted the slider, otherwise their adjustments would be lost. During startup I expected only a single call to `noise_suppress_update` so there would not be any blocking, but it seems it's possible for it to be called multiple times which I did not expect. I will try to figure out why this is happening.

Reply:
After some investigation, it turns out the CI build that @tt2468 tested with did not include the changes from this PR. After compiling an updated `obs-filters.dll` the changes work as expected.